packages/paste-core/components/badge/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,143 @@
+          color: 'colorTextNew',
+        });
+      });
+
+      it('should return the correct tokens when variant is "default"', () => {
+        expect(getVariantStyles('default')).toEqual({
+          backgroundColor: 'colorBackground',
+          color: 'colorTextWeak',
+        });
+      });
+    });
+  });
+
+  describe('render', () => {
+    it('should render correctly when variant is "default"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-1" variant="default">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-1');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextWeak');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackground');
+    });
+
+    it('should render correctly when variant is "new"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-2" variant="new">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-2');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextNew');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundNew');
+    });
+
+    it('should render correctly when variant is "info"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-3" variant="info">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-3');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextNeutral');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundNeutralWeakest');
+    });
+
+    it('should render correctly when variant is "warning"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-4" variant="warning">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-4');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextWarningStrong');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundWarningWeakest');
+    });
+
+    it('should render correctly when variant is "error"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-5" variant="error">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-5');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextError');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundErrorWeakest');
+    });
+
+    it('should render correctly when variant is "success"', () => {
+      const {getByTestId} = render(
+        <Badge data-testid="badge-6" variant="success">
+          test
+        </Badge>
+      );
+      const badgeElement = getByTestId('badge-6');
+
+      expect(badgeElement).toHaveStyleRule('color', 'colorTextSuccess');
+      expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundSuccessWeakest');
+    });
+  });
```

Comment:
I noticed that there aren't any tests covering the focusable variables here. Are they worth adding to ensure the button variant stays and does what a button should do?

Reply:
Yes, I think so. I omitted them because I wasn't confident that it was within scope based on the ticket description. I can add some now for both the anchor and the button variants.

I'll include:
- some interaction tests (prob with `testing-library/user-event` 
- some style checking tests for the focus and hover states

LMK if you'd like to see anything else. ✔️